core/src/main/java/cz/xtf/core/openshift/ClusterVersionInfo.java

```diff
@@ -21,8 +21,9 @@
 @Slf4j
 public class ClusterVersionInfo {
     // version must be in format major.minor.micro (4.8.13) or major.minor (4.8)
-    private static final Pattern versionPattern = Pattern
-            .compile("^(\\d+\\.\\d+)(\\.\\d+)?(-\\d+\\.nightly-\\d{4}-\\d{2}-\\d{2}-\\d{6})?$");
+    // consider any version after with 'x.y.z-abc' as nightly/developer build
+    // nightly in context of this class to not to change API
+    private static final Pattern versionPattern = Pattern.compile("^(\\d+\\.\\d+)(\\.\\d+)?(-.*)?$");
```

Comment:
There may be a problem as all non-standard versions are detected as dev-preview e.g. it will fail to download [4.7.0-rc.3](https://mirror.openshift.com/pub/openshift-v4/clients/ocp/4.7.0-rc.3/), which is in the classic stream.
Furthermore e.g. `99.99.99` is detected as a valid version, but that's happening even without this PR so I suppose that's expected behavior

Reply:
ok, updated, it leverages new `isDeveloperPreview()` method and added more tests